### PR TITLE
fix: Updated ButtonIcon disabled state

### PR DESCRIPTION
**Pull request by brianacnguyen**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**
- Update ButtonIcon's disabled state
- Also update ButtonIcon to probably disable

**Screenshots/Recordings**
Simulator Screen Shot - iPhone 14 Pro - 2023-06-29 at 10 08 19
Simulator Screen Shot - iPhone 14 Pro - 2023-06-29 at 10 08 22

https://github.com/MetaMask/metamask-mobile/assets/14355083/0d02ade2-7a43-4d97-b8bd-8ceee0bee5d1

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by georgewrmarshall** [comment]
Can this be closed in favor of https://github.com/MetaMask/metamask-mobile/pull/6727

**Comment by brianacnguyen** [comment]
> 

They were originally but I split them out to help get things in faster

**Review by tommasini — CHANGES_REQUESTED** [review]
Hey @brianacnguyen, left a question!

If possible could you do as well a recording of button icons working on the app, and a small description of the change? <3

File: app/component-library/components/Buttons/ButtonIcon/ButtonIcon.tsx [hunk 1]
```diff
@@ -20,35 +20,35 @@ import {
...
-  disabled,
   onPressIn,
   onPressOut,
   style,
   size = DEFAULT_BUTTON_ICON_SIZE,
   iconColorOverride = undefined,
+  isDisabled = false,
   ...props
 }: ButtonIconProps) => {
+  const [pressed, setPressed] = useState(false);
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply app/component-library/components/Buttons/ButtonIcon/ButtonIcon.tsx:31]: it is being set on `triggerOnPressedIn` and  `triggerOnPressedOut `

[the remark this answers is not in the document]
Comment (tommasini) [reply app/component-library/components/Buttons/ButtonIcon/ButtonIcon.tsx:31]: You are right! It's not new lines thought, that make me be confused, thanks!

**Comment by brianacnguyen** [comment]
> Hey @brianacnguyen, left a question!
> 
> If possible could you do as well a recording of button icons working on the app, and a small description of the change? <3

will add this soon

**Review by tommasini — APPROVED** [review]
LGTM!

**Pull request merged**